packages/blocks/src/api/raw-handling/utils.js

```diff
@@ -113,14 +113,41 @@ export function getBlockContentSchema( context ) {
 
 /**
  * Checks whether HTML can be considered plain text. That is, it does not contain
- * any elements that are not line breaks.
+ * any elements that are not line breaks, or it only contains a single non-semantic
+ * wrapper element (span) with no semantic child elements.
  *
  * @param {string} HTML The HTML to check.
  *
  * @return {boolean} Whether the HTML can be considered plain text.
  */
 export function isPlain( HTML ) {
-	return ! /<(?!br[ />])/i.test( HTML );
+	if ( ! /<(?!br[ />])/i.test( HTML ) ) {
+		return true;
+	}
+
+	const doc = document.implementation.createHTMLDocument( '' );
+	doc.body.innerHTML = HTML;
+
+	// Check if there's exactly one top-level element
+	if ( doc.body.children.length !== 1 ) {
+		return false;
+	}
+
+	const wrapper = doc.body.children.item( 0 );
+	const tagName = wrapper.tagName.toLowerCase();
+
+	// Only consider non-semantic wrapper tags
+	if ( tagName !== 'span' ) {
+		return false;
+	}
+
+	// Check if the wrapper contains only text nodes and <br> tags
+	// (no other semantic elements)
+	const hasSemanticChildren = Array.from(
+		wrapper.getElementsByTagName( '*' )
+	).some( ( el ) => el.tagName.toLowerCase() !== 'br' );
+
+	return ! hasSemanticChildren;
```

Comment:
At the risk of overthinking, we could:

* avoid the copy (`Array.from`) for a test where we want to fail early
* avoid lower-casing the tag name, in comformity with other parts of `raw-handling` (same goes for `const tagName` a few lines above)


```suggestion
	// Check if the wrapper contains only text nodes and <br> tags
	// (no other semantic elements)
	const descendants = element.getElementsByTagName( '*' );
	for ( let i = 0; i < descendants.length; i++ ) {
		if ( descendants.item( i ).tagName !== 'BR' ) {
			return false;
		}
	}
	
	return true;
```

Reply:
In the spirit of overthinking, you could probably go further and just look at `children`? Since you wouldn't have any nested children if you're only allowing `<br />` in the first place.


```suggestion
	// Check if the wrapper contains only text nodes and <br> tags
	// (no other semantic elements)
	for ( let i = 0; i < wrapper.children.length; i++ ) {
		if ( wrapper.children[ i ].tagName !== 'BR' ) {
			return false;
		}
	}

	return true;
```

Or let the browser do the work of filtering out text nodes and just look at the element children:


```suggestion
	let child = wrapper.firstElementChild;
	while ( child ) {
		if ( child.tagName !== 'BR' ) {
			return false;
		}

		child = child.nextElementSibling;
	}

	return true;
```

Reply:
Thank you for reviewing! Seems like there aren't any downsides so I committed your suggestion.